config/zfs-build.m4

```diff
@@ -27,7 +27,7 @@ AC_DEFUN([ZFS_AC_DEBUG], [
 	AC_MSG_CHECKING([whether assertion support will be enabled])
 	AC_ARG_ENABLE([debug],
 		[AS_HELP_STRING([--enable-debug],
-		[Enable assertion support @<:@default=no@:>@])],
+		[Enable assertion support and compiler features where available @<:@default=no@:>@])],
```

Comment:
The way you've implemented this setting `--enable-debug` has no effect of this compiler feature.  It only defines DEBUG and sets `-Werror`, so I'm assuming this is an unrelated bit of clarification.  In which case, how about the following message instead.

```
Enable assertions and -Werror @<:@default=no@:>@
```

Reply:
Actually it does. `-fsanitize` is only checked for and hence enabled when a debug build is performed. The whole new m4 block is guarded with
```m4
if test "$enable_debug" = "yes"
```
This allows a normal build to operate without `-fsanitize` even on systems which support it.

But I agree, this text needs rewording.

Reply:
Ahh, I see.  I glossed over that in my initial reading.  What do you think about moving the call points for these debug-only compiler checks in to `ZFS_AC_DEBUG_ENABLE`.  This way these M4 macro's will only run for debug builds where they're needed.

Reply:
I wanted to keep the stack size check together with this one for the sake of coherence and so that non-debug builds would still use a sane stack size check value. m4 is new to me so I'm sure my style is horrible.